### PR TITLE
ngMessages/Form Validation

### PR DESCRIPTION
**Pull request by alanlgirard**
Added ngMessages dependency so that our ng-messages show properly.  Now if you want multiple messages they will populate properly based on the criteria such as minlength or required, etc.

Added messages/form validation to both the login and the addIdea.  I limited the addIdea title to 4-30 characters and the description to 15-1000 characters.  I am open to suggestions if we want to alter some of these or get rid of them all together?

~~Re-factored the account page to display pertinent info.  Only implemented a "My Ideas: tab currently but left in place holders for "My Backs" and "My Teams" once the back-end is done being edited.  Right now there isn't an easy way to grab a user's backs/teams without looping through all of the ideas and checking.~~

@monotkate @Mctalian @YashdalfTheGray

### CODE REVIEW

File: src/ideas/addIdeaView/addIdeaView.tpl.html [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+                              md-maxlength="1000"
+                              ng-required></textarea>
+                    <div ng-messages="newIdeaForm.desc.$error" role="alert">
+                        <div ng-message="required">Required</div>
+                        <div ng-message="minlength">Your description is too short. Must be at least 15 characters</div>
+                        <div ng-message="maxlength">Your description is too long</div>
+                    </div>
                 </md-input-container>
-                <p></p>
+                </p></p>
```
Comment (Mctalian) [inline src/ideas/addIdeaView/addIdeaView.tpl.html]: Why 2 closing tags? (To be honest I never liked this random `<p></p>` to begin with, but why is it now `</p></p>`?

Can we use padding instead?

**Comment by Mctalian** [comment]
If you edit the idea and exceed the 1000 character limit, you can still submit the edits and the description disappears.

File: src/users/accountView/accountView.tpl.html [hunk 2]
```diff
@@ -1,12 +1,47 @@
...
-        <h2><center>{{user.name}} (<i>{{user.username}}</i>)</center></h2>
-        <h4><center>{{user.email}}</center><h4>
-    </div>
-    <md-button class="md-raised md-warn" ng-click="logout()" flex>Logout</md-button>
-</md-content>
+    </md-content>
+</md-card>
+<md-card>
+    <md-content>
+        <md-tabs md-dynamic-height md-border-bottom="">
```
Comment (Mctalian) [inline src/users/accountView/accountView.tpl.html]: I'm not really feeling the use of tabs here. I think we should try to follow more of a GitHub pattern and have different partitions on the screen where it shows "top 5" ideas, and maybe 5 most recent backs in another partition, and a "liked ideas" partition that will show the 5 "most popular" ideas that you've liked.

Some of this is more "future", and maybe we don't care for this commit. Or maybe people disagree and like the tabs, if that's the case, I'd like the ideas to not be listed straight down and look more like the All Ideas page instead.

Comment (davethenipper) [inline src/users/accountView/accountView.tpl.html]: That's a neat idea.  Some of the tabs are starting to get a little
overwhelming.

File: src/ideas/addIdeaView/addIdeaView.tpl.html (hunk 1 above)
Comment (alanlgirard) [inline src/ideas/addIdeaView/addIdeaView.tpl.html]: Left-over from messing around.  Removed and added padding in instead.

File: src/users/accountView/accountView.tpl.html (hunk 2 above)
Comment (alanlgirard) [inline src/users/accountView/accountView.tpl.html]: Yeah I didn't really like the tabs either but couldn't think of a good way to lay it out.  I do like the github profile page and will work on re-factoring the account page this weekend to have more of a partition look

**Comment by alanlgirard** [comment]
Reverted the account page for the time being while I work on it.  Want to get the ngMessages and form validation through first to not hold up anyone at all.  This PR should be good to be approved for the use of ngMessages on forms.

@Mctalian @monotkate @davethenipper @YashdalfTheGray

**Comment by monotkate** [comment]
I love the ellipses on a long title for the sidebar nav.  :100: 

On login, I kind of liked the required notifications for the color, but I suppose there's also assumptions on login.

For add/edit idea I'm a fan of adding the character limit either when you're clicking on the input itself, or always.  I don't like that it doesn't show things until you click on and then off.  It means you basically have no idea what you can input up to until you screw it up :(

**Comment by alanlgirard** [comment]
There is an issue with md-maxlength and ng-messages where the character count gets put in the ng-message (https://github.com/angular/material/issues/5853). I was hiding the message unless there was an error.  To fix this for the time being I changed the messages to show all of the time for the addIdea and editIdea pages.

**Comment by Mctalian** [comment]
Will this close any of the existing issues? Or does it only partially address one/some?

Can you tag appropriate issues?

**Comment by alanlgirard** [comment]
Closes issues #153, #175, #176, #177, and #179

**Comment by Mctalian** [comment]
Thank you for doing that.

I don't see any changes that would clear up #177, what is the character limit for comments?
And as far as #179, what about comments, backs, and updates?

**Comment by YashdalfTheGray** [comment]
I like this! Approved!

**Comment by alanlgirard** [comment]
Added validation for Updates, Comments, and Back description.  All are limited to 4-500 characters.  The back description is still not required but if they enter any text then it has to be within 4-500 for them to submit.

Should now fully close out #177 and #179 unless there are any other forms that I have missed.

**Comment by monotkate** [comment]
Approved!

**Comment by Mctalian** [comment]
:+1: Nice work!

**Comment by Mctalian** [comment]
a.k.a Approved

**Pull request merged**